### Step 1. 필요한 라이브러리 임포트

In [ ]:
import pandas as pd
import numpy as np

### Step 2. [데이터](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv) 확인

- order_id 주문번호
- quantity 주문 개수
- item_name 주문한 상품명
- choice_description 선택사항
- item_price 가격

### Step 3. 데이터 불러오기


**read_csv를 통해 데이터를 올바른 형태로 불러온 뒤 chipo에 저장해주세요**

In [ ]:
url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv'
    
chipo = pd.read_csv(url, sep='\t') # sep 변수연산자 종류 ( , | \t ) 

### Step 4. 초반 10행 조회

In [ ]:
chipo.head(10)


order_id  ...  item_price
0         1  ...      $2.39 
1         1  ...      $3.39 
2         1  ...      $3.39 
3         1  ...      $2.39 
4         2  ...     $16.98 
5         3  ...     $10.98 
6         3  ...      $1.69 
7         4  ...     $11.75 
8         4  ...      $9.25 
9         5  ...      $9.25 

[10 rows x 5 columns]

### Step 5. 데이터셋은 몇 개의 행으로 이루어져 있나요? (최소 2가지 이상 풀이법으로 풀어주세요)

4622

In [ ]:
# Solution 1
chipo.shape[0]
# entries <= 4622 observations

4622

In [ ]:
# Solution 2
chipo.info()
# entries <= 4622 observations

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


### Step 6. 데이터셋은 몇 개의 열로 이루어져 있나요?

5

In [ ]:
chipo.shape[1]

5

### Step 7. 열(column)의 이름들을 모두 조회해주세요

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

In [ ]:
chipo.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

### Step 8. 데이터셋의 인덱스를 조회해주세요

RangeIndex(start=0, stop=4622, step=1)

In [ ]:
chipo.index

RangeIndex(start=0, stop=4622, step=1)

### Step 9. 가장 많은 개수가 주문된 item_name은 무엇인가요?

Chicken Bowl	

In [ ]:
# chipo.groupby('item_name').sum()
# chipo.groupby('item_name').sum().sort_values(['quantity','order_id' ],ascending=False).head(1).index[0]
.index.index[0]

'Chicken Bowl'

### Step 10. 가장 많이 주문된 item은 몇 개 주문되었나요?
   힌트)  groupby 활용, item 총 주문 수량 확인 후 최상위로 정렬하여 결과 확인 
    
761

In [ ]:
c = chipo.groupby('item_name')["quantity"]
c = c.sum().max()
c
#c = c.sort_values(['quantity','order_id' ],ascending=False)
#c.head(1)

761

### Step 11. choice_description에서 가장 많이 주문된건 어떤 항목인가요?

Diet Coke 159

In [ ]:
c = chipo.groupby('choice_description').sum()
c = c.sort_values(['quantity'], ascending=False)
c.head(1).index[0]

'[Diet Coke]'

### Step 12. 총 몇 개의 item이 주문되었나요? (quantity의 총 합을 구해주세요)

4972

In [ ]:
chipo.quantity.sum()

4972

### Step 13. item price를 float로 변경해주세요

#### Step 13.a. item_price의 데이터타입을 조회해주세요

dtype('O')

In [ ]:
chipo.item_price.dtype

dtype('O')

#### Step 13.b. item_type에서 $기호를 제거하고 숫자 부분을 float으로 형변환 후 기존 dataframe에 저장해주세요

람다함수를 활용하면 편리합니다

In [ ]:
chipo.item_price.head()

0     $2.39 
1     $3.39 
2     $3.39 
3     $2.39 
4    $16.98 
Name: item_price, dtype: object

In [ ]:
dollarizer = lambda x : float(x[1:-1])
chipo.item_price = chipo.item_price.apply(dollarizer)  


TypeError: ignored

#### Step 13.c. 변경된 item_price의 데이터타입을 조회해주세요

dtype('float64')

In [ ]:
chipo.item_price.dtype

dtype('float64')

In [ ]:
chipo

order_id  ...  item_price
0            1  ...        2.39
1            1  ...        3.39
2            1  ...        3.39
3            1  ...        2.39
4            2  ...       16.98
...        ...  ...         ...
4617      1833  ...       11.75
4618      1833  ...       11.75
4619      1834  ...       11.25
4620      1834  ...        8.75
4621      1834  ...        8.75

[4622 rows x 5 columns]

### Step 14. 총 매출은 얼마인가요?

39237.02

In [ ]:
# 총 매출 = quantity x item_price

revenue = (chipo['quantity'] * chipo['item_price']).sum()
revenue

# revenue was : $39237.02

print("revenue was : $" + str(np.round(revenue, 2)))

revenue was : $39237.02


39237.02

### Step 15. 총 몇 건의 주문이 발생헀나요? (주문번호 기준) 

1834

**4가지 해답을 적어뒀습니다. 이 해답을 참조하여 Step 17을 해결해주세요**

In [ ]:
# Solution 1 : len() , unique()
# len(chipo['order_id'].unique())
len(chipo.order_id.unique())

1834

In [ ]:
# Solution 2 :shape
chipo.order_id.unique().shape

(1834,)

In [ ]:
# Solution 3 : nunique()
chipo.order_id.nunique()

1834

In [ ]:
# Solution 4 : count()
chipo.order_id.value_counts().count()

1834

In [ ]:
chipo

order_id  ...  item_price
0            1  ...        2.39
1            1  ...        3.39
2            1  ...        3.39
3            1  ...        2.39
4            2  ...       16.98
...        ...  ...         ...
4617      1833  ...       11.75
4618      1833  ...       11.75
4619      1834  ...       11.25
4620      1834  ...        8.75
4621      1834  ...        8.75

[4622 rows x 5 columns]

### Step 16. 주문당 평균 매출은 얼마인가요?

21.394231188658654

In [ ]:
# Solution 1 : groupby() 미사용
chipo['revenue'] = chipo['quantity'] * chipo['item_price']
order_grouped = chipo.groupby(by=['order_id']).sum()
order_grouped.mean()['revenue']

21.394231188658654

In [ ]:
# Solution 2 : groupby() 사용
# chipo.groupby(by=['order_id']).sum().mean()
a = chipo.groupby(by=['order_id']).sum().mean()['revenue']
print("주문당 평균 매출: $" + str(np.round(a, 3)))

주문당 평균 매출: $21.394


### Step 17. 몇 종류의 item이 판매되었나요?

50

In [ ]:
chipo.item_name.value_counts()

Chicken Bowl                             726
Chicken Burrito                          553
Chips and Guacamole                      479
Steak Burrito                            368
Canned Soft Drink                        301
Steak Bowl                               211
Chips                                    211
Bottled Water                            162
Chicken Soft Tacos                       115
Chicken Salad Bowl                       110
Chips and Fresh Tomato Salsa             110
Canned Soda                              104
Side of Chips                            101
Veggie Burrito                            95
Barbacoa Burrito                          91
Veggie Bowl                               85
Carnitas Bowl                             68
Barbacoa Bowl                             66
Carnitas Burrito                          59
Steak Soft Tacos                          55
6 Pack Soft Drink                         54
Chips and Tomatillo Red Chili Salsa       48
Chicken Cr

In [ ]:
# chipo.item_name.value_counts()
chipo.item_name.value_counts().count()

50